addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1814,14 +1860,25 @@ var SnippetsMenu = Widget.extend({
             editor.removeSnippet();
         });
     },
+    /**
+     * @private
+     */
+    _onSnippetCloned: function (ev) {
+        if (ev.data.$target.hasClass('o_snippet_invisible')) {
+            this._addInvisibleElementsToLeftPanel(ev.data.$target);
+        }
```

Comment:
That className should not be repeated everywhere. I would do this:
- Rename `_addInvisibleElementsToLeftPanel` into `_updateInvisibleDOM`
- The function do not receive the invisible elements to add but the container in which to search for invisible elements (using `getEditableArea` by default).

Reply:
What would be the container in this case? the snippet itself? so same param

Reply:
Yep, just avoiding repeating the condition. (If I implemented the feature I would probably have started the opposite way of yours: on update empty everything and re-search all the invisible elements, but maybe it is overkill in this case).